scripts/includes.sh

```diff
@@ -83,6 +84,55 @@ function send_mail_content() {
     fi
 }
 
+########################################
+# Export variables from .env file.
+# Arguments:
+#     None
+# Outputs:
+#     variables with prefix 'DOTENV_'
+# Reference:
+#     https://gist.github.com/judy2k/7656bfe3b322d669ef75364a46327836
+########################################
+function export_env_file() {
+    if [[ -f "${ENV_FILE}" ]]; then
+        color blue "find \"${ENV_FILE}\" file and export variables"
+        set -a
+        source <(cat "${ENV_FILE}" | sed -e '/^#/d;/^\s*$/d' -e 's/\(\w*\)[ \t]*=[ \t]*\(.*\)/DOTENV_\1=\2/')
+        set +a
+    fi
+}
+
+########################################
+# Get variables from
+#     environment variables,
+#     secret file in environment variables,
+#     secret file in .env file,
+#     environment variables in .env file.
+# Arguments:
+#     variable name
+# Outputs:
+#     variable value
+########################################
+function get_env() {
+    local VAR="$1"
+    local VAR_FILE="${VAR}_FILE"
+    local VAR_DOTENV="DOTENV_${VAR}"
+    local VAR_DOTENV_FILE="DOTENV_${VAR_FILE}"
+    local VALUE=""
+
+    if [[ "${!VAR:-}" ]]; then
```

Comment:
what does the `!` do?

Reply:
This means a reference to a variable, which you can see in https://stackoverflow.com/questions/2634590/using-a-variable-to-refer-to-another-variable-in-bash

Reply:
Thank you for the clarification. That example you listed shows an indirect reference where the real variable name is the value of a different variable. 

As far as I can tell we don't have a second variable which holds the reference.
```
if [[ -n "$VAR" ]]; then
```
This previous example should do the exact same thing and is (by my opinion) more readable.

Reply:
Thank you for clarification. 

I'd love to se a `-n` in the if/elseif statement.